selfdrive/car/honda/carstate.py

```diff
@@ -260,18 +260,20 @@ def update(self, cp, cp_cam, cp_body):
     else:
       ret.cruiseState.speed = cp.vl["CRUISE"]["CRUISE_SPEED_PCM"] * CV.KPH_TO_MS
 
-    self.brake_switch = cp.vl["POWERTRAIN_DATA"]["BRAKE_SWITCH"] != 0
     if self.CP.carFingerprint in HONDA_BOSCH_ALT_BRAKE_SIGNAL:
       ret.brakePressed = cp.vl["BRAKE_MODULE"]["BRAKE_PRESSED"] != 0
     else:
       # brake switch has shown some single time step noise, so only considered when
       # switch is on for at least 2 consecutive CAN samples
-      # panda safety only checks BRAKE_PRESSED signal
-      ret.brakePressed = bool(cp.vl["POWERTRAIN_DATA"]["BRAKE_PRESSED"] or
-                              (self.brake_switch and self.brake_switch_prev and cp.ts["POWERTRAIN_DATA"]["BRAKE_SWITCH"] != self.brake_switch_prev_ts))
-
-      self.brake_switch_prev = self.brake_switch
-      self.brake_switch_prev_ts = cp.ts["POWERTRAIN_DATA"]["BRAKE_SWITCH"]
+      # brake switch rises earlier than brake pressed but is never 1 when in park
+      brake_switch_vals = cp.updated["POWERTRAIN_DATA"]["BRAKE_SWITCH"]
+      brake_switch = cp.vl["POWERTRAIN_DATA"]["BRAKE_SWITCH"] != 0
+      if len(brake_switch_vals):
+        if len(brake_switch_vals) > 1:
+          self.brake_switch_prev = brake_switch_vals[-2] != 0
+        self.brake_switch_active = brake_switch and self.brake_switch_prev
```

Comment:
If we reset active to False each time, then cases where we don't receive an update (so we don't know if brake switch is 0 or 1), but check was previously True, we would fall to 0. Panda maintains its previous `brakePressed` value if it doesn't receive a message.